htmdx/cli.py

```diff
@@ -1,182 +1,146 @@
 
+    # Send the registration data
+    url = "https://www.acellera.com/licensing/htmd/check.php"
+    res = requests.post(url, data, timeout=10)
 
-def check_registration(product=None):
-    if not product:
-        product = "NA"
+    # Check the response
+    if res.status_code == 200:
+        status = res.json()
+        if "approved" in status:
+            return True
 
-    reg_file = os.path.join(
-        os.path.expanduser("~"), ".htmd", ".registered-htmd", "registration"
-    )
+        # Note: remove after switching to the new backend
+        if "pending" in status:
+            return True
+
+    print('Registration is not approved: %s' % res.text)
+    return False
 
-    if not (os.path.isfile(reg_file) and check_approval(product, reg_file)):
-        accept_license(product=product)
-        do_register(product=product)
 
+def _accept_licence(product):
 
-def accept_license(product=None):
-    show_license(product=product)
-    print("Type 'yes' to accept the license [no] : ", end="")
-    sys.stdout.flush()
-    ret = input().strip()
+    ret = input("Type 'yes' to accept the license [no] : ").strip()
     if ret != "yes":
-        os._exit(1)
+        sys.exit(1)
 
 
-def show_license(product=None):
-    path = os.path.abspath(
-        os.path.dirname(os.path.dirname(inspect.getfile(accept_license)))
-    )
-    if product:
-        path = os.path.join(path, product)
+def _show_licence(product):
 
-    path = os.path.join(path, "LICENCE.txt")
+    # Find the licence file
+    path = os.path.join(os.path.dirname(__file__), '..', product, "LICENCE.txt")
     if not os.path.exists(path):
-        print("Could not find license file. Aborting")
-        os._exit(1)
+        print("Cannot find the licence file!")
+        sys.exit(1)
 
-    print(sys.stdin)
+    # Show the licence file
     if sys.stdout.isatty() and sys.stdin:
         if platform.system() == "Windows":
             call(["c:\\Windows\\System32\\more.com", path])
         else:
             call(["more", path])
     else:
-        fh = open(path, "r")
-        print(fh.read())
-        fh.close()
+        with open(path) as fh:
+            print(fh.read())
+
+
+def htmd_registration(product="htmd"):
 
+    if not _check_registration(product):
+        _show_licence(product)
+        _accept_licence(product)
+        htmd_register(product)
 
-def do_register(product=None):
-    name = ""
-    institution = ""
-    email = ""
-    city = ""
-    country = ""
+
+def _ask(prompt):
+
+    value = ""
+    while value == "":
+        value = input(prompt).strip()
+
+    return value
+
+
+def htmd_register(product="htmd"):
 
     print(
-        "\n Welcome to HTMD. We'd like to know who you are so that we can keep in touch!"
-    )
-    print(" Please enter your name, affiliation and institutional email address.\n")
-    print(" Please provide valid information so that it might be verified.\n")
-
-    while email == "" or not ("@" in email) or not ("." in email):
```

Comment:
It was move to the server side: https://github.com/Acellera/licensing/pull/26/files#diff-8de0b206346441414a0593f1fc67d9008051f21e7529e32b94c58c26de79f53dR35

Reply:
Ah ok, it will be a bit more annoying for users who fake their emails since they will have to start the registration from scratch but that's their own problem :D

Reply:
Anyway, I have to check on the server to prevent some garbage going into the database.